coral-component-masonry/src/scripts/Masonry.js

```diff
@@ -313,6 +335,118 @@ class Masonry extends BaseComponent(HTMLElement) {
+
+   Setting the attribute to other than allowed values will fallback to {@link MasonryAriaGridEnum.OFF}.
+
+   @type {String}
+   @default {@link MasonryAriaGridEnum.OFF}
+   @htmlattribute ariagrid
+   @htmlattributereflected
+   */
+  get ariaGrid() {
+    return this._ariaGrid || ariaGrid.OFF;
+  }
+  set ariaGrid (value) {
+    value = transform.string(value);
+
+    // Ensure correct values
+    if (value !== ariaGrid.ON && value !== ariaGrid.OFF) {
+      console.warn('Coral.Masonry: Unsupported ariaGrid value: ', value, '. Will fallback to ', ariaGrid.OFF);
+      value = ariaGrid.OFF;
+    }
+
+    //update current state
+    this._ariaGrid = value;
+    this._reflectAttribute('ariagrid', this._ariaGrid);
+
+    // Update role for this masonry
+    if (this._ariaGrid === ariaGrid.ON) {
+      // Preserve existing role and set new role
+      this._preservedAriaRole = this.getAttribute('role');
+      this.setAttribute('role', 'row');
+    }
+    else if (this._ariaGrid == ariaGrid.OFF) {
+      // Restore or remove role
+      if (this._preservedAriaRole) {
+        this.setAttribute('role', this._preservedAriaRole);
+      }
+      else {
+        this.removeAttribute('role');
+      }
+    }
+
+    // Update parent and child item roles based on current state
+    this._updateAriaRoleForParent(this._ariaGrid);
+    this._updateAriaColumnCountForParent(this._ariaGrid);
+    this._updateAriaRoleForItems(this._ariaGrid);
+  }
+
+  /** @private */
+  _updateAriaRoleForParent(activateAriaGrid) {
+    if (!this.parentNode) {
+      return;
+    }
+
+    if (activateAriaGrid === ariaGrid.ON) {
+      // Save/set role for the parent as grid
+      this._preservedParentAriaRole = this.parentNode.getAttribute('role');
+      this.parentNode.setAttribute('role', 'grid');
+    }
+    else {
+      // Restore/remove role of the parent element
+      if (this._preservedParentAriaRole) {
+        this.parentNode.setAttribute('role', this._preservedParentAriaRole);
+      }
+      else {
+        this.parentNode.removeAttribute('role');
+      }
+
+      // Remove aria-colcount
+      this.parentNode.removeAttribute('aria-colcount');
+    }
+  }
+
+  /** @private */
+  _updateAriaRoleForItems(activateAriaGrid) {
+    let columnIndex = 1;
+    this.items.getAll().forEach((item) => {
+      this._updateAriaRoleForItem(item, columnIndex++, activateAriaGrid);
+    });
+  }
+
+  /** @private */
+  _updateAriaRoleForItem(item, columnIndex, activateAriaGrid) {
+    if (activateAriaGrid === ariaGrid.ON) {
+      item.setAttribute('role','gridcell');
+      item.setAttribute('aria-colindex', columnIndex);
+    }
+    else {
+      item.removeAttribute('role');
+      item.removeAttribute('aria-colindex');
+    }
+  }
+
+  /** @private */
+  _updateAriaColumnCountForParent(activateAriaGrid) {
+    if (activateAriaGrid === ariaGrid.ON) {
+      this.parentNode.setAttribute('aria-colcount', this.items.length);
```

Comment:
`parentNode` may not exist, maybe we should add a check as your did for _updateAriaRoleForParent()

Reply:
Added parent node check.